core/server/worker/src/main/java/alluxio/worker/block/CacheRequestManager.java

```diff
@@ -67,32 +70,63 @@
    * @param blockWorker handler to the block worker
    * @param fsContext context
    */
-  public AsyncCacheRequestManager(ExecutorService service, BlockWorker blockWorker,
+  public CacheRequestManager(ExecutorService service, BlockWorker blockWorker,
       FileSystemContext fsContext) {
-    mAsyncCacheExecutor = service;
+    mCacheExecutor = service;
     mBlockWorker = blockWorker;
     mFsContext = fsContext;
-    mPendingRequests = new ConcurrentHashMap<>();
-    mLocalWorkerHostname =
-        NetworkAddressUtils.getLocalHostName(
-            (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
+    mActiveCacheRequests = new ConcurrentHashMap<>();
+    mLocalWorkerHostname = NetworkAddressUtils.getLocalHostName(
+        (int) ServerConfiguration.getMs(PropertyKey.NETWORK_HOST_RESOLUTION_TIMEOUT_MS));
   }
 
   /**
-   * Handles a request to cache a block asynchronously. This is a non-blocking call.
+   * Handles a request to cache a block. If it's async cache, it wouldn't throw exception.
    *
-   * @param request the async cache request fields will be available
+   * @param request the cache request fields will be available
    */
-  public void submitRequest(AsyncCacheRequest request) {
-    ASYNC_CACHE_REQUESTS.inc();
+  public void submitRequest(CacheRequest request) throws AlluxioException, IOException {
+    CACHE_REQUESTS.inc();
     long blockId = request.getBlockId();
-    if (mPendingRequests.putIfAbsent(blockId, request) != null) {
+    if (mActiveCacheRequests.putIfAbsent(blockId, request) != null) {
       // This block is already planned.
-      ASYNC_CACHE_DUPLICATE_REQUESTS.inc();
+      LOG.debug("request already planned: {}", request);
       return;
     }
+    boolean async = request.getAsync();
+    if (!async) {
+      CACHE_REQUESTS_SYNC.inc();
+      submitSyncRequest(request, blockId);
+    } else {
+      CACHE_REQUESTS_ASYNC.inc();
+      submitASyncRequest(request, blockId);
```

Comment:
Submitting part is the same for both. Exception handling no need to be different until waiting for result, which is only relevant for sync request. So for both async and sync, we can still submit and handle pool rejections in a unified manner.

Once passed that, so the job is in the pool, then there can be follow up logic specific to sync calls, where you can handle ExecutionException and InterruptedException.

Reply:
I thought RejectedExecutionException handling would be different since async just log but sync cache tend to throw an exception. Updated to a unified manner